### PR TITLE
[INTERNAL] Resource#getProject: Add notice for extensions

### PR DESCRIPTION
**Pull request by RandomByte**

### CODE REVIEW

File: lib/Resource.js [hunk 1]
```diff
@@ -330,6 +330,13 @@ class Resource {
 
 	/**
 	 * Retrieve the project assigned to the resource
+	 * <br/>
+	 * <b>Note for UI5 Tooling extensions (i.e. custom tasks, custom middleware):</b>
+	 * In order to ensure compatibility across UI5 Tooling versions, consider using the
+	 * <code>getProject(resource)</code> methods provided by
+	 * [TaskUtil]{@link module:@ui5/project/build/helpers/TaskUtil} and
+	 * [MiddlewareUtil]{@link module:@ui5/server.middleware.MiddlewareUtil}, which will
+	 * return a Specification Version-compatible Project interface.
```
Comment (flovogt) [inline lib/Resource.js:339]: in other PRs you have used the term `Specification Version-dependent`. IMHO we should use only one of the terms

Comment (flovogt) [reply lib/Resource.js:339]: @KlattG any preferred variant?

**Review by flovogt — CHANGES_REQUESTED** [review]

File: lib/Resource.js (hunk 1 above)
Comment (KlattG) [reply lib/Resource.js:339]: @flovogt I have no problem with using both terms. 'compatible' sounds ideal here.

**Review by KlattG — APPROVED** [review]

File: lib/Resource.js [hunk 2]
```diff
@@ -330,6 +330,13 @@ class Resource {
 
 	/**
 	 * Retrieve the project assigned to the resource
+	 * <br/>
+	 * <b>Note for UI5 Tooling extensions (i.e. custom tasks, custom middleware):</b>
+	 * In order to ensure compatibility across UI5 Tooling versions, consider using the
+	 * <code>getProject(resource)</code> methods provided by
```
Comment (matz3) [inline lib/Resource.js:336]:
```suggestion
	 * <code>getProject(resource)</code> method provided by
```
It's only one method, right?

File: lib/Resource.js [hunk 3]
```diff
@@ -330,6 +330,13 @@ class Resource {
 
 	/**
 	 * Retrieve the project assigned to the resource
+	 * <br/>
+	 * <b>Note for UI5 Tooling extensions (i.e. custom tasks, custom middleware):</b>
+	 * In order to ensure compatibility across UI5 Tooling versions, consider using the
+	 * <code>getProject(resource)</code> method provided by
```
Comment (flovogt) [inline lib/Resource.js:336]: methods

**Review by flovogt — CHANGES_REQUESTED** [review]

File: lib/ResourceFacade.js [hunk 4]
```diff
@@ -187,6 +187,13 @@ class ResourceFacade {
 
 	/**
 	 * Retrieve the project assigned to the resource
+	 * <br/>
+	 * <b>Note for UI5 Tooling extensions (i.e. custom tasks, custom middleware):</b>
+	 * In order to ensure compatibility across UI5 Tooling versions, consider using the
+	 * <code>getProject(resource)</code> methods provided by
```
Comment (flovogt) [inline lib/ResourceFacade.js:193]: method

**Review by flovogt — CHANGES_REQUESTED** [review]

File: lib/ResourceFacade.js (hunk 4 above)
Comment (RandomByte) [inline lib/ResourceFacade.js:193]:
```suggestion
	 * <code>getProject(resource)</code> method provided by
```

**Review by flovogt — APPROVED** [review]
LGTM

**Pull request merged**